core/drivers/crypto/crypto_api/acipher/ecc.c

```diff
@@ -570,6 +580,7 @@ TEE_Result drvcrypt_asym_alloc_ecc_public_key(struct ecc_public_key *key,
 
 		switch (type) {
 		case TEE_TYPE_SM2_PKE_PUBLIC_KEY:
+		case TEE_TYPE_SM2_DSA_PUBLIC_KEY:
 			key->curve = TEE_ECC_CURVE_SM2;
```

Comment:
Why isn't `ecc->alloc_publickey()` taking care of this?

Reply:
ecc->alloc_publickey() can not get key_type(such as TEE_TYPE_SM2_DSA_PUBLIC_KEY) to decide which curve it is.

Reply:
`crypto_asym_alloc_ecc_keypair()` in `core/lib/libtomcrypt/ecc.c` and `lib/libmbedtls/core/ecc.c` does that.

It might be worth adding a comment here about why some special treatment is needed.